sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -319,14 +362,19 @@ final class DataFrameWriter private[sql](df: DataFrame) {
         checkpointPath.toUri.toString
       }
 
-      val sink = new MemorySink(df.schema)
+      if (!Seq(OutputMode.Append, OutputMode.Complete).contains(outputMode)) {
```

Comment:
The tricky thing is that we want to make the Memory Sink compatible with update internally, but we may not want public API to support update mode yet.